src/target/page/TargetPage.tsx

```diff
@@ -28,13 +54,42 @@ const TargetDetailsSection = ({
   title: string;
   children: ReactNode;
 }): JSX.Element => (
-  <section id={id} className="target-details-section">
+  <TargetSection id={id}>
     <Collapse isOpen collapseContentId={`${id}-section`} title={title}>
       {children}
     </Collapse>
-  </section>
+  </TargetSection>
 );
 
+const TargetSourceLink = ({
+  id,
+  target,
+}: {
+  id: string;
+  target?: Target;
+}): JSX.Element | null => {
+  const url: string | undefined = getTargetServiceAdminUrl(target?.links);
+
+  if (!url) {
+    return null;
+  }
+
+  return (
+    <TargetSection id={id}>
+      <p>
+        Toimipisteeseen liittyvät tiedot kieliversioineen ovat lähtöisin
+        Toimipisterekisteristä. Jos haluat muuttaa tietoja, se on mahdollista
```

Comment:
Would maybe simplify this sentence 'Jos haluat muuttaa tietoja, se on mahdollista Toimipisterekisterissä' -> 'Tietojen muuttaminen on mahdollista Toimipisterekisterissä'

Reply:
I just followed the design, but yes, it felt a bit complex for me too. I can iterate it a bit, and let's see how it goes.